src/mesa/main/get.c

```diff
@@ -74,8 +74,8 @@
                                 (GLint) ((F) * 65536.0f) )
 
 #define INT_TO_BOOLEAN(I)     ( (I) ? GL_TRUE : GL_FALSE )
-#define INT_TO_FIXED(I)       ( ((I) > SHRT_MAX) ? INT_MAX : \
-                                ((I) < SHRT_MIN) ? INT_MIN : \
+#define INT_TO_FIXED(I)       ( ((GLint)(I) > SHRT_MAX) ? INT_MAX : \
```

Comment:
I think you should do this just for the one case where we have warning (for the glenum16), so do the cast in row 3005, then this should be ok for upstream. For upstreaming, I recommend making 2 separate patches, one for log changes and other for this one.